lib/crystalball/map_generator/factory_bot_strategy.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+require 'crystalball/map_generator/factory_bot_strategy/factory_gem_loader'
+
+Crystalball::MapGenerator::FactoryBotStrategy::FactoryGemLoader.require!
+
+require 'crystalball/map_generator/base_strategy'
+require 'crystalball/map_generator/helpers/path_filter'
+require 'crystalball/map_generator/factory_bot_strategy/dsl_patch'
+require 'crystalball/map_generator/factory_bot_strategy/factory_runner_patch'
+
+module Crystalball
+  class MapGenerator
+    # Map generator strategy to include list of strategies which was used in an example.
+    class FactoryBotStrategy
+      include ::Crystalball::MapGenerator::BaseStrategy
+      include ::Crystalball::MapGenerator::Helpers::PathFilter
+
+      class << self
+        def factory_bot_constant
+          defined?(::FactoryBot) ? ::FactoryBot : ::FactoryGirl
+        end
+
+        # List of factories used by current example
+        #
+        # @return [Array<String>]
+        def used_factories
+          @used_factories ||= []
+        end
+
+        # Map of factories to files
+        #
+        # @return [Hash<String, String>]
+        def factory_definitions
+          @factory_definitions ||= {}
+        end
+
+        # Reset cached list of factories
+        def reset_used_factories
+          @used_factories = []
+        end
+      end
+
+      def after_register
+        DSLPatch.apply!
+        FactoryRunnerPatch.apply!
+      end
+
+      # Adds factories related to the spec to the case map
+      # @param [Crystalball::CaseMap] case_map - object holding example metadata and affected files
+      def call(case_map, _)
+        self.class.reset_used_factories
+        yield case_map
+        case_map.push(*filter(self.class.used_factories.flat_map { |f| self.class.factory_definitions[f] }.compact))
```

Comment:
what's the case when we have no factory definition but the factory was used?

Reply:
When it was used by an alias. Now I solved that case